lib/Service/RecordingService.php

```diff
@@ -139,42 +142,120 @@ public function store(Room $room, string $owner, array $file): void {
+			return;
+		}
+
+		$task = new Task(
+			AudioToText::ID,
+			['input' => $fileNode->getId()],
+			Application::APP_ID,
+			$owner,
+			'call/transcription/' . $room->getToken(),
+		);
+
 		try {
-			$this->speechToTextManager->scheduleFileTranscription($fileNode, $owner, Application::APP_ID);
-			$this->logger->debug('Scheduled transcription of call recording');
-		} catch (PreConditionNotMetException $e) {
-			// No Speech-to-text provider installed
-			$this->logger->debug('Could not generate transcript of call recording', ['exception' => $e]);
-		} catch (InvalidArgumentException $e) {
-			$this->logger->warning('Could not generate transcript of call recording', ['exception' => $e]);
+			$this->taskProcessingManager->scheduleTask($task);
+			$this->logger->debug('Scheduled call recording transcript');
+		} catch (Exception $e) {
+			$this->logger->error('An error occurred while trying to transcribe the call recording', ['exception' => $e]);
 		}
 	}
 
-	public function storeTranscript(string $owner, File $recording, string $transcript): void {
+	/**
+	 * @param 'transcript'|'summary' $aiTask
+	 */
+	public function storeTranscript(string $owner, string $roomToken, int $recordingFileId, string $output, string $aiTask): void {
+		$userFolder = $this->rootFolder->getUserFolder($owner);
+		$recordingNodes = $userFolder->getById($recordingFileId);
+
+		if (empty($recordingNodes)) {
+			$this->logger->warning("Could not save recording $aiTask as the recording could not be found", [
+				'owner' => $owner,
+				'roomToken' => $roomToken,
+				'recordingFileId' => $recordingFileId,
+			]);
+			throw new InvalidArgumentException('owner_participant');
+		}
+		$recording = array_pop($recordingNodes);
 		$recordingFolder = $recording->getParent();
-		$roomToken = $recordingFolder->getName();
+
+		if ($recordingFolder->getName() !== $roomToken) {
+			$this->logger->warning("Could not determinate conversation when trying to store $aiTask of call recording, as folder name did not match customId conversation token");
+			throw new InvalidArgumentException('owner_participant');
+		}
 
 		try {
 			$room = $this->roomManager->getRoomForUserByToken($roomToken, $owner);
 			$participant = $this->participantService->getParticipant($room, $owner);
 		} catch (ParticipantNotFoundException) {
-			$this->logger->warning('Could not determinate conversation when trying to store transcription of call recording');
+			$this->logger->warning("Could not determinate conversation when trying to store $aiTask of call recording");
 			throw new InvalidArgumentException('owner_participant');
 		}
 
-		$transcriptFileName = pathinfo($recording->getName(), PATHINFO_FILENAME) . '.md';
+		$shouldTranscribe = $this->serverConfig->getAppValue('spreed', 'call_recording_transcription', 'no') === 'yes';
+		$shouldSummarize = $this->serverConfig->getAppValue('spreed', 'call_recording_summary', 'no') === 'yes';
```

Comment:
check this first so you can return early if both features are turned off

Reply:
Well either is true, otherwise we wouldn't be in storeTranscript (which could be renamed to storeAITextOutput now), because we only end up here, if a task got resolved after `store()` checked that either is true already.